### PR TITLE
Adds plugin build-name-setter to weekly-bom

### PR DESCRIPTION
**Pull request by damianszczepanik**
### Testing done

```[tasklist]
### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue
```

### CODE REVIEW

**Comment by MarkEWaite** [comment]
@damianszczepanik thanks for the pull request.  I've pushed two additional changes that are needed in order to compile the code.  I also ran the tests with the command:

```bash
PLUGINS=build-name-setter  bash local-test.sh
```

The tests fail on my computer when run with that command.  When tests fail with that command, it usually indicates that the tests interact unexpectedly with the other plugins that are loaded in the bill of materials.

The test failure messages that I see are recorded in a [gist](https://gist.github.com/MarkEWaite/41918b9013da8baecc2415f46cc55d7d)

The short form of the failure message is:

```
[ERROR] Failures: 
[ERROR]   BuildNameSetterTest.shouldExpand_JOB_NAME_full_env_macro:40->assertDisplayName:85 expected:<[b_bar]> but was:<[#1]>
[ERROR]   BuildNameSetterTest.shouldExpand_JOB_NAME_macro_and_JOB_NAME_full_env_macro:70->assertDisplayName:85 expected:<[d_built-in_foo]> but was:<[#1]>
[ERROR]   BuildNameSetterTest.shouldUse_default_config_values_if_null:80->assertDisplayName:85 expected:<[foo]> but was:<[#1]>
```

**Review by MarkEWaite — CHANGES_REQUESTED** [review]
Tests in the plugin likely need to be updated to deal with other plugins being loaded at the same time.

**Comment by damianszczepanik** [comment]
@MarkEWaite I have rebased the PR, fixed plugin version and builds are passing now

**Comment by MarkEWaite** [comment]
> @MarkEWaite I have rebased the PR, fixed plugin version and builds are passing now

Thanks.  When I try to run the updated pull request with the command:

```bash
PLUGINS=build-name-setter bash local-test.sh
```

I still see reports of test failures like this:

```
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
Running tests for org.jenkins-ci.plugins:build-name-setter:2.4.2-SNAPSHOT
[INFO] Running org.jenkinsci.plugins.buildnamesetter.BuildNameSetterTest
[INFO] Running InjectedTest
[INFO] Tests run: 12, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 16.96 s -- in InjectedTest
[ERROR] Tests run: 6, Failures: 3, Errors: 0, Skipped: 0, Time elapsed: 33.27 s <<< FAILURE! -- in org.jenkinsci.plugins.buildnamesetter.BuildNameSetterTest
[ERROR] org.jenkinsci.plugins.buildnamesetter.BuildNameSetterTest.shouldExpand_JOB_NAME_macro_and_JOB_NAME_full_env_macro -- Time elapsed: 17.43 s <<< FAILURE!
org.junit.ComparisonFailure: expected:<[d_built-in_foo]> but was:<[#1]>
        at org.junit.Assert.assertEquals(Assert.java:117)
        at org.junit.Assert.assertEquals(Assert.java:146)
        at org.jenkinsci.plugins.buildnamesetter.BuildNameSetterTest.assertDisplayName(BuildNameSetterTest.java:85)
        at org.jenkinsci.plugins.buildnamesetter.BuildNameSetterTest.shouldExpand_JOB_NAME_macro_and_JOB_NAME_full_env_macro(BuildNameSetterTest.java:70)
        at java.base/java.lang.reflect.Method.invoke(Method.java:580)
        at org.jvnet.hudson.test.JenkinsRule$1.evaluate(JenkinsRule.java:656)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
        at java.base/java.lang.Thread.run(Thread.java:1583)

[ERROR] org.jenkinsci.plugins.buildnamesetter.BuildNameSetterTest.shouldUse_default_config_values_if_null -- Time elapsed: 2.560 s <<< FAILURE!
org.junit.ComparisonFailure: expected:<[foo]> but was:<[#1]>
        at org.junit.Assert.assertEquals(Assert.java:117)
        at org.junit.Assert.assertEquals(Assert.java:146)
        at org.jenkinsci.plugins.buildnamesetter.BuildNameSetterTest.assertDisplayName(BuildNameSetterTest.java:85)
        at org.jenkinsci.plugins.buildnamesetter.BuildNameSetterTest.shouldUse_default_config_values_if_null(BuildNameSetterTest.java:80)
        at java.base/java.lang.reflect.Method.invoke(Method.java:580)
        at org.jvnet.hudson.test.JenkinsRule$1.evaluate(JenkinsRule.java:656)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
        at java.base/java.lang.Thread.run(Thread.java:1583)

[ERROR] org.jenkinsci.plugins.buildnamesetter.BuildNameSetterTest.shouldExpand_JOB_NAME_full_env_macro -- Time elapsed: 3.513 s <<< FAILURE!
org.junit.ComparisonFailure: expected:<[b_bar]> but was:<[#1]>
        at org.junit.Assert.assertEquals(Assert.java:117)
        at org.junit.Assert.assertEquals(Assert.java:146)
        at org.jenkinsci.plugins.buildnamesetter.BuildNameSetterTest.assertDisplayName(BuildNameSetterTest.java:85)
        at org.jenkinsci.plugins.buildnamesetter.BuildNameSetterTest.shouldExpand_JOB_NAME_full_env_macro(BuildNameSetterTest.java:40)
        at java.base/java.lang.reflect.Method.invoke(Method.java:580)
        at org.jvnet.hudson.test.JenkinsRule$1.evaluate(JenkinsRule.java:656)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
        at java.base/java.lang.Thread.run(Thread.java:1583)

[INFO]
[INFO] Results:
[INFO]
[ERROR] Failures:
[ERROR]   BuildNameSetterTest.shouldExpand_JOB_NAME_full_env_macro:40->assertDisplayName:85 expected:<[b_bar]> but was:<[#1]>
[ERROR]   BuildNameSetterTest.shouldExpand_JOB_NAME_macro_and_JOB_NAME_full_env_macro:70->assertDisplayName:85 expected:<[d_built-in_foo]> but was:<[#1]>
[ERROR]   BuildNameSetterTest.shouldUse_default_config_values_if_null:80->assertDisplayName:85 expected:<[foo]> but was:<[#1]>
[INFO]
[ERROR] Tests run: 18, Failures: 3, Errors: 0, Skipped: 0
```

File: sample-plugin/pom.xml [hunk 1]
```diff
@@ -33,6 +33,11 @@
       <groupId>org.jenkins-ci.plugins</groupId>
       <artifactId>authorize-project</artifactId>
     </dependency>
+    <dependency>
+      <groupId>org.jenkins-ci.plugins</groupId>
+      <artifactId>build-name-setter</artifactId>
+      <version>2.4.2-rc239.def8c27a_4446</version>
```
Comment (MarkEWaite) [inline sample-plugin/pom.xml:39]: Version is not needed in this pom file as far as I understand.
```suggestion
```

**Comment by damianszczepanik** [comment]
Fixing tests to pass the build won't be as easy so closing this PR for now.

**Pull request closed without merging**